pkg/minishift/clusterup/clusterup.go

```diff
@@ -64,14 +65,39 @@ type ClusterUpConfig struct {
 func ClusterUp(config *ClusterUpConfig, clusterUpParams map[string]string, runner util.Runner) error {
 	cmdArgs := []string{"cluster", "up", "--use-existing-config"}
 
+	// Deal with extra flags
+	var extraFlags []string
+	if val, ok := clusterUpParams[startFlags.ExtraClusterUpFlags.Name]; ok {
+		extraFlags = strings.Fields(val)
+		delete(clusterUpParams, startFlags.ExtraClusterUpFlags.Name)
+	}
+
+	// Check if clusterUp flags are supported
 	for key, value := range clusterUpParams {
 		if !oc.SupportFlag(key, config.OcPath, runner) {
+			println(key)
 			return errors.New(fmt.Sprintf("Flag %s is not supported for oc version %s. Use 'openshift-version' flag to select a different version of OpenShift.", key, config.OpenShiftVersion))
 		}
+
 		cmdArgs = append(cmdArgs, "--"+key)
 		cmdArgs = append(cmdArgs, value)
 	}
 
+	// Check if extra flags are supported
+	for _, extraFlag := range extraFlags {
+		isFlag := strings.HasPrefix(extraFlag, "--")
+		key := strings.Trim(extraFlag, "-")
+
+		if !isFlag {
```

Comment:
on second thought, I think the problem is to split in general. I don't think a regular expression will help. How would you handle/distinguish things like "--loglevel 1 --service-catalog" from  "--service-catalog f-loglevel 1" from "--service-catalog --foo". If you want to handle this properly you need command line processing. IMO we would be better off to pass the extra arguments through as they are. This way we can support any form of extra argument. It is basically up to the user to decide and know what to pass through.